mdsuite/file_io/extxyz_files.py

```diff
@@ -0,0 +1,305 @@
+            n_configs_float = num_lines / (self._n_particles + self.n_header_lines)
+            n_configs = int(round(n_configs_float))
+            assert abs(n_configs_float - n_configs) < 1e-10
+
+        # same properties for all species
+        properties_list = []
+        for prop_name, prop_col_idxs in self._properties_dict.items():
+            properties_list.append(
+                mdsuite.database.simulation_database.PropertyInfo(
+                    name=prop_name, n_dims=len(prop_col_idxs)
+                )
+            )
+        species_list = []
+        for sp_name, sp_indices_dict in self._species_dict.items():
+            species_list.append(
+                mdsuite.database.simulation_database.SpeciesInfo(
+                    name=sp_name,
+                    n_particles=len(sp_indices_dict["line_idxs"]),
+                    properties=properties_list,
+                )
+            )
+
+        self._mdata = mdsuite.database.simulation_database.TrajectoryMetadata(
+            n_configurations=n_configs,
+            box_l=box_l,
+            sample_rate=sample_rate,
+            species_list=species_list,
+        )
+        self._batch_size = mdsuite.utils.meta_functions.optimize_batch_size(
+            filepath=self.file_path, number_of_configurations=n_configs
+        )
+
+        return self._mdata
+
+    def get_configurations_generator(
+        self,
+    ) -> typing.Iterator[mdsuite.file_io.file_read.TrajectoryChunkData]:
+        n_configs = self._mdata.n_configurations
+        n_batches, n_configs_remainder = divmod(int(n_configs), int(self._batch_size))
+
+        with open(self.file_path, "r") as file:
+            file.seek(0)
+            for _ in tqdm.tqdm(range(n_batches)):
+                yield self._read_process_n_configurations(file, self._batch_size)
+            if n_configs_remainder > 0:
+                yield self._read_process_n_configurations(file, n_configs_remainder)
+
+    def _read_process_n_configurations(
+        self, file, n_configs
+    ) -> mdsuite.file_io.file_read.TrajectoryChunkData:
+        """
+        Read n_configs configurations and bring them to the structore needed for the yield of get_configurations_generator()
+        Parameters
+        ----------
+        file
+            The open trajectory file. Note: Calling this function will advance the reading point in the file
+        n_configs
+
+        Returns
+        -------
+
+        """
+        chunk = mdsuite.file_io.file_read.TrajectoryChunkData(
+            self._mdata.species_list, n_configs
+        )
+
+        for config_idx in range(n_configs):
+            # skip the header
+            mdsuite.file_io.file_read.skip_n_lines(file, self.n_header_lines)
+            # read one config
+            traj_data = np.stack(
+                [
+                    np.array(list(file.readline().split()))
+                    for _ in range(self._n_particles)
+                ]
+            )
```

Comment:
the inner `np.array` is probably not required. Speedwise there is no difference though 
![image](https://user-images.githubusercontent.com/46721498/144587665-a6c6a7fa-770c-4698-89fe-2200e88dfe2f.png)